src/include/migraphx/op/allocate.hpp

```diff
@@ -36,20 +36,53 @@ namespace op {
 struct allocate
 {
     shape s{};
+    // for dynamic allocate to set the buffer type
+    shape::type_t buf_type = shape::half_type;
+    // flag for dynamic allocate to use shape attr rather than [0, max] ranges in compute_shape()
+    // useful for optimizations later on
+    bool use_shape_attr = false;
+
     template <class Self, class F>
     static auto reflect(Self& self, F f)
     {
-        return pack(f(self.s, "shape"));
+        return pack(f(self.s, "shape"),
+                    f(self.buf_type, "buf_type"),
+                    f(self.use_shape_attr, "use_shape_attr"));
     }
+
     std::string name() const { return "allocate"; }
+
     shape compute_shape(const std::vector<shape>& inputs) const
     {
-        migraphx::check_shapes{inputs, *this, true}.has(0);
-        return s;
+        migraphx::check_shapes{inputs, *this, true}.has(0, 1);
+        if(inputs.empty() or use_shape_attr)
+        {
+            return s;
+        }
+        else
+        {
+            const auto& out_dims = inputs.at(0);
```

Comment:
```suggestion
            const auto& out_shape = inputs.at(0);
```
It is of type shape not dims.